research/object_detection/exporter_lib_v2.py

```diff
@@ -75,11 +97,33 @@ def _run_inference_on_images(self, image):
 class DetectionFromImageModule(DetectionInferenceModule):
   """Detection Inference Module for image inputs."""
 
-  @tf.function(
-      input_signature=[
-          tf.TensorSpec(shape=[1, None, None, 3], dtype=tf.uint8)])
-  def __call__(self, input_tensor):
-    return self._run_inference_on_images(input_tensor)
+  def __init__(self, detection_model,
+               use_side_inputs=False,
+               zipped_side_inputs=None):
+    """Initializes a module for detection.
+
+    Args:
+      detection_model: The detection model to use for inference.
+
+    """
+    self.side_input_names = []
+    sig = [tf.TensorSpec(shape=[1, None, None, 3], dtype=tf.uint8)]
+    if use_side_inputs:
+      for info in zipped_side_inputs:
+        self.side_input_names.append(info[2])
+        sig.append(tf.TensorSpec(shape=info[0],
+                                 dtype=info[1],
+                                 name=info[2]))
+
+    def __call__(input_tensor, *side_inputs):
+      kwargs = dict(zip(self.side_input_names.split(","), side_inputs))
```

Comment:
as discussed offline, let's remove split.

Reply:
Just curious, what would you recommend is used instead?